app/controllers/api_controller/arbitration_profiles.rb

```diff
@@ -0,0 +1,43 @@
+class ApiController
+  module ArbitrationProfiles
+    def create_resource_arbitration_profiles(_type, _id, data)
+      validate_profile_data(data)
+      parse_set_provider(data)
```

Comment:
Ah, I see. Thanks! I think though that we could improve on this:

* This method receives some data, passes it to `parse_set_provider`, which then mutates the data. This kind of thing can be hard to debug, so I'd suggest `dup`ing or otherwise creating a new hash without mutating the one that's passed in
* If `parse_set_provider` is parsing and setting, then it's doing two things. WDYT about it just doing the parsing:

```ruby
attributes = {}
attributes["ext_management_system"] = parse_provider(data)
attributes["availability_zone"] = parse_availability_zone(data)
# ...
ArbitrationProfile.create(attributes)
```